src/Raven.Studio/typescript/models/database/index/indexFieldOptions.ts

```diff
@@ -520,11 +519,21 @@ class indexFieldOptions {
         });
     }
 
+    addCustomAnalyzers(customAnalyzers: string[]) {
+        customAnalyzers.forEach(name => {
+            if (!this.analyzersNamesDictionary().find(x => x.studioName === name)) {
+                const customAnalyzerEntry = { studioName: name, serverName: name } as analyzerName;
+                this.analyzersNamesDictionary.push(customAnalyzerEntry);
+                this.analyzersNamesDictionary().sort((a, b) => a.studioName > b.studioName ? 1 : -1);
```

Comment:
no need to sort that in every loop step. 
Assign values to array. 
```
const analyzers = this.analyzersNamesDictionary();
foreach (...) { analyzers.push(...)
analyzers.sort();

// assign back to observable:
this.analyzersNamesDictionary(analyzers)
```

Reply:
done